### ISSUE TITLE
Subsequent datetime.now() calls return the same datetime

### ISSUE THREAD

**Issue by spacemanspiff2007** [python/cpython#124832]
# Bug report

### Bug description:

Subsequent `datetime.now()` may return the same moment in time.

```python
from datetime import datetime, UTC

now1 = datetime.now(UTC)
for i in range(10_000):
    a = i
now2 = datetime.now(UTC)

assert now1 < now2
```
```
    assert now1 < now2
           ^^^^^^^^^^^
AssertionError
```

This does seem to work as expected:
```python
from time import sleep
from datetime import datetime, UTC

now1 = datetime.now(UTC)
sleep(0.000_000_001)
now2 = datetime.now(UTC)

assert now1 < now2
```

I'm not sure if this is a bug, but at least it's a documentation issue.
There's nothing in the [docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.now) explicitly indicating this behavior.

### CPython versions tested on:

3.12

### Operating systems tested on:

Windows

### Linked PRs
* gh-124834
* gh-125145
* gh-125146

**Comment by vstinner** [comment]
> I'm not sure if this is a bug, but at least it's a documentation issue.

It's not a bug.

> There's nothing in the [docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.now) explicitly indicating this behavior.

What do you expect in the doc?

**Comment by spacemanspiff2007** [comment]
I understand and suspected as much.

> What do you expect in the doc?

A warning that due to the clock resolution it's possible that two `datetime.now()` calls can return the same time even though a "substantial" amount of time has passed.

I had some tests failing which check that a datetime value which was updated was in the future.
That failed and I could not figure out why. I supplied `UTC` as tz because the docs mention it might increase precision but with no success.
A warning or hint would have been nice which I tried to add in the PR.
